### ISSUE TITLE
Is there a decoder ckpt also?

### ISSUE THREAD

**Issue by xunmengshe2x** [ZhengyaoFang/RS-STE#5]
Hi there, 

 is there a decoder ckpt file also? Getting errors when trying to run this. Have the rsste_finetune.ckpt but i think the decoder ckpt is missing?

**Comment by ZhengyaoFang** [comment]
Hi! I have just tested and updated the code for inference today. The decoder ckpt is also in rsste_finetune.ckpt, which is called "vqgan" and will be transfered into "decoder" in model/model.py line 80.

**Comment by xunmengshe2x** [comment]
Thank you for that. 
 So i got it running but for MAIL example I see this (so it should have replaced MAIL with HELLO) 



![Image](https://github.com/user-attachments/assets/3fc686b4-641d-4080-b12d-420be0dc5769)


This is code I am using for inference ( i am using this within a different application), used 768 to get it working



`

# Create annotation file
    annotation_data = {
        "image1_paths": [input_path],
        "image2_paths": [],
        "image1_rec": ["MAIL"],  # Will be recognized by the model
        "image2_rec": [target_text]  # Target text to edit into the image
    }
    annotation_file = os.path.join(annotation_dir, "temp_inference.pkl")
    with open(annotation_file, "wb") as f:
        pickle.dump(annotation_data, f)

    # Set up output path
    output_path = os.path.join(output_dir, f"result_{datetime.now().strftime('%Y%m%d_%H%M%S')}.png")

    try:
        # Add the repository to the Python path
        sys.path.append(repo_dir)

        # Import necessary modules
        from main import instantiate_from_config, get_obj_from_str

        # Load model configurations from the cloned repository
        vqgan_config = os.path.join(repo_dir, "configs/vqgan_decoder.yaml")
        transformer_config = os.path.join(repo_dir, "configs/synth_pair.yaml")
        checkpoint_path = os.path.join(model_dir, "rsste-finetune.ckpt")

        decoder_config = OmegaConf.load(vqgan_config)
        config = OmegaConf.load(transformer_config)
        
        # Set the decoder config and checkpoint path
        config.model.params.decoder_config = decoder_config.model
        config.model.params.ckpt_path = checkpoint_path
        
        # FIX: Explicitly set n_embd to 768 to match the checkpoint dimensions
        config.model.params.n_embd = 768
        config.model.params.transformer_config.params.n_embd = 768

        # Change from 768 to 384 to match the config file
        #config.model.params.n_embd = 384
        #config.model.params.transformer_config.params.n_embd = 384

        
        # Initialize model
        model = instantiate_from_config(config.model).to('cuda')
        model.eval()

        # Create dataset for the single image
        from data.dataset import InferenceDataset
        from torch.utils.data import DataLoader
        dataset = InferenceDataset(config.data.params.validation.params.size, annotation_file)
        dataloader = DataLoader(dataset, batch_size=1, shuffle=False)

        # Run inference
        with torch.no_grad():
            for batch in dataloader:
                img1 = batch["image1"].to('cuda')
                img1 = img1.permute(0, 3, 1, 2).to(memory_format=torch.contiguous_format)
                if img1.dtype == torch.double:
                    img1 = img1.float()

                img1_quant_latent_ = model.conv(img1).flatten(2).permute(0, 2, 1)
                rec2_indices, _ = model.str_converter.encode(batch["rec2"])
                rec2_indices = rec2_indices.to(img1_quant_latent_.device)
                rec2_embd = model.str_embd(rec2_indices)
                rec1_mask = model.masked_rec_embd.weight[0][None, None, :].expand(rec2_indices.shape[0], rec2_indices.shape[1], -1)
                img2_mask = model.masked_img_imbd.weight[0][None, None, :].expand(img1_quant_latent_.shape[0], img1_quant_latent_.shape[1], -1)
                inputs = torch.cat([rec2_embd, img1_quant_latent_, rec1_mask, img2_mask], dim=1)
                embeddings, logits = model.transformer(inputs)
                rec1_indices = torch.topk(logits[:,288:320,:], k=1, dim=-1)[1].view(batch["image1"].shape[0],-1)
                pred_rec = model.str_converter.decode(rec1_indices)
                img2_rec_quant_latent = model.conv_o(embeddings[:,320:576,:].permute(0, 2, 1).contiguous().view(embeddings.shape[0], -1, 8, 32))
                img2_rec = model.decoder.decode(img2_rec_quant_latent)
                edited_img = crop_and_resize(img2_rec, batch["image1_size"], batch['ori_size'], 0)
                Image.fromarray(edited_img).save(output_path)

        # Read the result image
        with open(output_path, "rb") as f:
            result_image_bytes = f.read()

        # Encode the result image as base64
        result_base64 = base64.b64encode(result_image_bytes).decode("utf-8")

        # Return the result
        return {
            "original_text": pred_rec[0] if pred_rec else "Unknown",
            "edited_text": target_text,
            "result_image_base64": result_base64
        }




`



if use 384 instead of 768 i get mismatch errors. 


Do you  know what I am doing wrong to not get the correct clear inference result? 

Thank you!

**Comment by ZhengyaoFang** [comment]
Hello! Sorry, after troubleshooting, the problem was found to be due to the wrong checkpoint upload😭. The download link for checkpoint has been updated, so please re-download it

**Comment by xunmengshe2x** [comment]
> Hello! Sorry, after troubleshooting, the problem was found to be due to the wrong checkpoint upload😭. The download link for checkpoint has been updated, so please re-download it

Hi, thank you for that!! :) 

It works now! 

Thank you!!

![Image](https://github.com/user-attachments/assets/62eac500-9141-459f-9664-05d3197c2609)
